Modules/Core/Tests/CoreTests/JSON/JSONTests.swift

```diff
@@ -28,208 +29,291 @@ public class JSONTests : XCTestCase {
+        parser = JSONMapParser(stream: parserStream)
+        XCTAssertThrowsError(try parser.parse(deadline: .never))
         buffer = Buffer("\"\\ud83")
-        XCTAssertThrowsError(try parser.parse(buffer))
+        parserStream = BufferStream(buffer: buffer)
+        parser = JSONMapParser(stream: parserStream)
+        XCTAssertThrowsError(try parser.parse(deadline: .never))
         buffer = Buffer("\"\\ud83d")
-        XCTAssertThrowsError(try parser.parse(buffer))
+        parserStream = BufferStream(buffer: buffer)
+        parser = JSONMapParser(stream: parserStream)
+        XCTAssertThrowsError(try parser.parse(deadline: .never))
         buffer = Buffer("\"\\ud83d\\")
-        XCTAssertThrowsError(try parser.parse(buffer))
+        parserStream = BufferStream(buffer: buffer)
+        parser = JSONMapParser(stream: parserStream)
+        XCTAssertThrowsError(try parser.parse(deadline: .never))
         buffer = Buffer("\"\\ud83d\\u")
-        XCTAssertThrowsError(try parser.parse(buffer))
+        parserStream = BufferStream(buffer: buffer)
+        parser = JSONMapParser(stream: parserStream)
+        XCTAssertThrowsError(try parser.parse(deadline: .never))
         buffer = Buffer("\"\\ud83d\\ud")
-        XCTAssertThrowsError(try parser.parse(buffer))
+        parserStream = BufferStream(buffer: buffer)
+        parser = JSONMapParser(stream: parserStream)
+        XCTAssertThrowsError(try parser.parse(deadline: .never))
         buffer = Buffer("\"\\ud83d\\ude")
-        XCTAssertThrowsError(try parser.parse(buffer))
+        parserStream = BufferStream(buffer: buffer)
+        parser = JSONMapParser(stream: parserStream)
+        XCTAssertThrowsError(try parser.parse(deadline: .never))
         buffer = Buffer("\"\\ud83d\\ude0")
-        XCTAssertThrowsError(try parser.parse(buffer))
+        parserStream = BufferStream(buffer: buffer)
+        parser = JSONMapParser(stream: parserStream)
+        XCTAssertThrowsError(try parser.parse(deadline: .never))
         buffer = Buffer("\"\\ud83d\\ude0e")
-        XCTAssertThrowsError(try parser.parse(buffer))
+        parserStream = BufferStream(buffer: buffer)
+        parser = JSONMapParser(stream: parserStream)
+        XCTAssertThrowsError(try parser.parse(deadline: .never))
         buffer = Buffer("\"\\ud83d\\u0000")
-        XCTAssertThrowsError(try parser.parse(buffer))
+        parserStream = BufferStream(buffer: buffer)
+        parser = JSONMapParser(stream: parserStream)
+        XCTAssertThrowsError(try parser.parse(deadline: .never))
         buffer = Buffer("\"\\u0000\\u0000")
-        XCTAssertThrowsError(try parser.parse(buffer))
+        parserStream = BufferStream(buffer: buffer)
+        parser = JSONMapParser(stream: parserStream)
+        XCTAssertThrowsError(try parser.parse(deadline: .never))
         buffer = Buffer("\"\\u0000\\ude0e")
-        XCTAssertThrowsError(try parser.parse(buffer))
+        parserStream = BufferStream(buffer: buffer)
```

Comment:
I think this test case could really use a few helper functions to clean up all the cruft.